frb_codegen/src/ir/func.rs

```diff
@@ -19,7 +20,10 @@ impl IrFunc {
 /// Represents a function's output type
 #[derive(Debug, Clone)]
 pub enum IrFuncOutput {
-    ResultType(IrType),
+    ResultType {
+        result: IrType,
```

Comment:
uber-nit: From https://doc.rust-lang.org/std/result/, shall we rename `result` to `ok`, `value`, or sth like that